core/cog-gamepad.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * cog-gamepad.c
+ * Copyright (C) 2022 Igalia S.L.
+ *
+ * Distributed under terms of the MIT license.
+ */
+
+#include "cog.h"
+#include <wpe/wpe.h>
+
+#if COG_ENABLE_GAMEPAD_MANETTE
+extern const struct wpe_gamepad_provider_interface manette_provider_interface;
+extern const struct wpe_gamepad_interface          manette_device_interface;
+#endif
+
+static const struct CogGamepadBackend {
+    const char                                  *name;
+    const struct wpe_gamepad_provider_interface *provider;
+    const struct wpe_gamepad_interface          *device;
+} s_gamepad_backends[] = {
+#if COG_ENABLE_GAMEPAD_MANETTE
+    {
+        "manette",
+        &manette_provider_interface,
+        &manette_device_interface,
+    },
+#endif
+    {
+        "disabled",
+        NULL,
+        NULL,
+    },
+};
+
+/* selected gamepad backend name */
+static const char *backend;
+/* selected gamepad backend interfaces */
+static const struct CogGamepadBackend *selected;
+/* selected gamepad backend provider interfaces */
+static struct wpe_gamepad_provider_interface provider_interface;
```

Comment:
I kept the global `static struct wpe_gamepad_provider_interface provider_interface;` otherwise the compiler complains with: `error: initializaer element is not a compile-time constant'